packages/deploy/src/validator.ts

```diff
@@ -13,17 +13,19 @@ export function parseAndValidate(input: string): {
   doc: ProjectSpec;
 } {
   let errors: Error[] = [];
-  let keys: string[] = [];
+  let workflowKeys: Set<string> = new Set();
+  let jobKeys: Set<string> = new Set();
   const doc = YAML.parseDocument(input);
 
-  function pushUniqueKey(context: YAML.Pair, key: string) {
-    if (keys.includes(key)) {
+  function pushUniqueKey(context: YAML.Pair, key: string, keyType: 'workflow' | 'job') {
```

Comment:
Instead of using two top-level variables and selecting them based on as string, can we just pass the correct collection into this function?
```
function pushUniqueKey(context: YAML.Pair, key: string, arr: string[]) {
  ...
}
```

So `pushUniqueKey` means: "add a key to THIS array BUT only if it doesn't exist"

This change also makes the function a bit more pure, which is nice.

Once you've done this, I also think you should declare the two arrays further down, inside the  validation loop. I'll comment where I mean.

Reply:
Thanks for the suggestion! Simplifying pushUniqueKey by removing keyType makes it more focused. Declaring arrays inside validateWorkflows enhances organization. I initially added keyType for clearer error messages to users.